### PR TITLE
Add option to use a URL parameter for cache-busting instead of copying files

### PR DESCRIPTION
**Pull request by PLPeeters**
This PR adds an option named `useUrlParam` that makes it possible to cache-bust using a URL parameter instead of copying files to add a version number directly to the filename. It replaces references like `/path/to/asset.js` with `/path/to/asset.js?cache-bust=HASH` without touching the file.

### CODE REVIEW

**Comment by hollandben** [comment]
Hi @PLPeeters. The recent bump to version `1.x.x` has removed the functionality for cache busting using a query string. Have a read through the documentation and issue #147 for more info

**Comment by PLPeeters** [comment]
Hi, 

Reading through the issue, I see why it would make sense to prefer cache busting by rewriting files entirely. However, in my case, my build process is set up in a way that prevents me from using that method. 

I would change my build process entirely if I could, but that's something I can't do for now, because it would require changing the server config for multiple projects. 

In the end, why not leave the choice of the cache busting method up to the end user? I totally see why you wouldn't want to bloat the code with unnecessary options, but I think this one makes sense and could be a relatively common use case. 

Anyway, I've made my case, the rest is up to you :)

**Comment by hollandben** [comment]
I'm not again giving people the option. With the new release I wanted to trim the fat and then see what people noticed was missing. I've left some comments so after they're done I'm happy to add the feature back in

File: tasks/cachebust.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ var DEFAULT_OPTIONS = {
     jsonOutput: false,
     jsonOutputFilename: 'grunt-cache-bust.json',
     length: 16,
-    separator: '.'
+    separator: '.',
+    useUrlParam: false
```
Comment (hollandben) [inline tasks/cachebust.js]: Can you change the option name to `queryString`

File: tasks/cachebust.js [hunk 2]
```diff
@@ -80,10 +83,14 @@ module.exports = function() {
         }
 
         function addFileHash(str, hash, separator) {
-            var parsed = url.parse(str);
-            var ext = path.extname(parsed.pathname);
+            if (opts.useUrlParam) {
+                return str + '?cache-bust=' + hash;
```
Comment (hollandben) [inline tasks/cachebust.js]: Don't believe the `cache-bust` section is needed, so can we remove that and just have the hash

**Comment by PLPeeters** [comment]
Sure thing. I hesitated regarding the named parameter too, guess I'll just leave the hash then.

**Comment by hollandben** [comment]
A few more things to do:

- [ ] Add unit tests for `queryString` option
- [ ] Update documentation and change log
- [ ] Bump version number in `package.json` file

**Comment by PLPeeters** [comment]
On it.

**Comment by PLPeeters** [comment]
@hollandben Any particular reason this hasn't been merged yet? :)

**Comment by hollandben** [comment]
Nope - I hadn't seen/been notified of the commits tbh. Will have a look shortly

**Comment by PLPeeters** [comment]
I guessed that might have been the case, hence my comment :)

File: README.md [hunk 3]
```diff
@@ -72,7 +72,7 @@ The `src` part of the configuration you should have seen before as it's used by
 // Here is a short summary of the options and some of their 
 defaults. Extra details are below.
 {
-    algorithm: 'md5',                             // Algoirthm used for hashing files
+    algorithm: 'md5',                             // Algorithm used for hashing files
```
Comment (hollandben) [inline README.md]: Haha thanks

**Comment by hollandben** [comment]
Nice - thanks

File: README.md (hunk 3 above)
Comment (PLPeeters) [inline README.md]: Happened to come across it :D

**Comment by hollandben** [comment]
Have publish `v1.2.0` - I bumped it a minor version as it was a new feature for this major release

**Pull request merged**